### ISSUE TITLE
Allow stashing unstaged changes only

### ISSUE THREAD

**Issue by michael-hawker** [ForkIssues/TrackerWin#444]
Sometimes I only want to stash part of my set of files.  It'd be great to be able to stash a set of files, or just the unstaged ones or something.  I've seen other git clients like SourceTree do this as well.

**Comment by DanPristupov** [comment]
> I've seen other git clients like SourceTree do this as well.

I'm not familiar with it. Can you provide more details or show a screenshot at least?

**Comment by michael-hawker** [comment]
The feature Source Tree has is documented [here](https://confluence.atlassian.com/sourcetreekb/stash-a-file-with-sourcetree-785332122.html).

Their stash dialog has a checkbox that lets you keep the staged files in the workspace:

![](https://confluence.atlassian.com/sourcetreekb/files/785332122/785332100/1/1445927259388/stash2.png)

I do really like Fork's `Stage New Files` option as well, but it'd be great to get the alternate mode to stash and keep staged files.  It's super helpful for squirreling away some working parts of an environment.

**Comment by DanPristupov** [comment]
OK, it's the `--keep-index` option.

Not sure if we should introduce it. New (untracked) unstaged files will not be stashed. As for me, this can lead to a quite confusing result: staged changes remain staged, some unstaged files are stashed, another part of unstaged files are still there.

**Comment by michael-hawker** [comment]
@DanPristupov yeah, but you're not always adding new files all the time.  I think it in Fork, if they're two radio options that you can't both choose at once, then it'd mitigate that confusion.  You can do one or the other, but not both.

**Comment by STeeL835** [comment]
What about stashing staged files? I'm probably doing something wrong, but I often find myself using `git stash save -p` command from console in need to exclude some changes to test code without them.

**Comment by DanPristupov** [comment]
> What about stashing staged files?

The original is about stashing staged **changes**, which is different.

> `git stash save -p`

This shows all changes, both staged and unstaged. Not sure how it can help us.

**Comment by carun** [comment]
> As for me, this can lead to a quite confusing result: staged changes remain staged, some unstaged files are stashed, another part of unstaged files are still there.

@DanPristupov I agree with this to some extent. But lack of such an option makes stash not quite useful. Is it possible to bring an option to stash the indexed changes only (inverse of `--keep-index`)? This will make it seamless to move around the indexed changes between branches.

**Comment by albertodeago** [comment]
@DanPristupov what about implementing somehow the git stash -p interactive functionality? it let you choose what to stash and what not. It's great imho.
It's one of the few command I find myself to use outside of fork

**Comment by Srekel** [comment]
I want this feature quite often. I make some changes that I want to keep (but perhaps not commit yet), some changes that I want to stash for later (maybe to delete, but I want to keep them for a bit). 

I would like:
1. Select the files in unstashed you want to stash
2. Stash selected files
3. Those files should get stashed _with the diff shown in the unstashed mode_. Their staged changes should stay staged.

**Comment by carun** [comment]
Fork 1.98.0.0 allows us to select all the files in the **Unstaged** area and stash only them without impacting the **Staged** area. Isn't that enough?
1. Select any file in the Unstaged area and press Ctrl-A.
2. Right click and click on Stash X files
![image](https://github.com/fork-dev/TrackerWin/assets/471374/e45209f0-c60b-4658-9a08-1c17bb0338e9)
3. Final result
![image](https://github.com/fork-dev/TrackerWin/assets/471374/329418e9-856a-40ed-9b65-c0211dec12b4)

**Comment by Srekel** [comment]
No, I would like for only the unstaged changes to be stashed. I try to do this on occasion forgetting that's not the way it works and then I have to unstash and re-stage the changes I wanted.
<img width="277" alt="image" src="https://github.com/fork-dev/TrackerWin/assets/1578924/b80db2a6-10d8-41b4-b75f-60562d17d653">

**Comment by carun** [comment]
@Srekel Ah, I think that message is incorrect. If you follow the steps mentioned above, it will stage **only unstaged** changes.
